analytics/msf4j-analytics/src/main/java/org/wso2/msf4j/analytics/httpmonitoring/config/HTTPMonitoringConfigBuilder.java

```diff
@@ -15,29 +15,61 @@
 import org.wso2.msf4j.analytics.httpmonitoring.config.model.HTTPMonitoringConfig;
-import org.yaml.snakeyaml.Yaml;
+import org.wso2.msf4j.analytics.internal.DataHolder;
 
-import java.util.Optional;
+import java.nio.file.Files;
+import java.nio.file.Paths;
 
 /**
  * Build {@link HTTPMonitoringConfig} from the YAML file
  */
 public final class HTTPMonitoringConfigBuilder {
 
+    private static final Logger logger = LoggerFactory.getLogger(HTTPMonitoringConfigBuilder.class);
+    private static final String DEPLOYMENT_YAML_SYS_PROPERTY = "msf4j.conf";
+
     public static HTTPMonitoringConfig build() {
-        Optional<String> metricsConfigFileContent = Utils.readFile("http-monitoring.conf", "http-monitoring.yml");
-        if (metricsConfigFileContent.isPresent()) {
+        ConfigProvider configProvider = DataHolder.getInstance().getConfigProvider();
+        HTTPMonitoringConfig configurationObject = null;
+        if (configProvider == null) {
+            if (DataHolder.getInstance().getBundleContext() != null) {
+                throw new RuntimeException(
+                        "Failed to populate HTTPMonitoringConfig Configuration. Config Provider is Null.");
+            }
+            //Standalone mode
+            String deploymentYamlPath = System.getProperty(DEPLOYMENT_YAML_SYS_PROPERTY);
+            if (deploymentYamlPath == null || deploymentYamlPath.isEmpty()) {
+                logger.info("System property '" + DEPLOYMENT_YAML_SYS_PROPERTY +
+                            "' is not set. Default deployment.yaml file will be used.");
+                deploymentYamlPath = HTTPMonitoringConfig.class.getResource("/deployment.yaml").getPath();
+            } else if (!Files.exists(Paths.get(deploymentYamlPath))) {
+                String msg = "Couldn't find " + deploymentYamlPath;
+                throw new RuntimeException(msg);
+            }
+
             try {
-                Yaml yaml = new Yaml();
-                return yaml.loadAs(metricsConfigFileContent.get(), HTTPMonitoringConfig.class);
-            } catch (RuntimeException e) {
-                throw new RuntimeException("Failed to populate HTTP Monitoring Configuration", e);
+                configProvider = ConfigProviderFactory.getConfigProvider(Paths.get(deploymentYamlPath), null);
+                DataHolder.getInstance().setConfigProvider(configProvider);
+            } catch (ConfigurationException e) {
+                throw new RuntimeException("Error loading deployment.yaml Configuration", e);
             }
-        } else {
-            return new HTTPMonitoringConfig();
         }
+
+        try {
+            configurationObject =
+                    DataHolder.getInstance().getConfigProvider().getConfigurationObject(HTTPMonitoringConfig.class);
+        } catch (ConfigurationException e) {
+            logger.error("Error loading configurations from deployment.yaml", e);
+            configurationObject = new HTTPMonitoringConfig();
```

Comment:
rather than just logging the error here, shouldn't we be throwing a runtime exception here if configuration loading is failing ?